backends-common/postgres/src/main/java/org/apache/james/backends/postgres/quota/PostgresQuotaCurrentValueDAO.java

```diff
@@ -54,35 +52,49 @@ public PostgresQuotaCurrentValueDAO(@Named(DEFAULT_INJECT) PostgresExecutor post
     }
 
     public Mono<Void> increase(QuotaCurrentValue.Key quotaKey, long amount) {
-        return postgresExecutor.executeVoid(dslContext -> Mono.from(dslContext.insertInto(TABLE_NAME)
-            .set(IDENTIFIER, quotaKey.getIdentifier())
-            .set(COMPONENT, quotaKey.getQuotaComponent().getValue())
-            .set(TYPE, quotaKey.getQuotaType().getValue())
-            .set(CURRENT_VALUE, amount)
-            .onConflictOnConstraint(PRIMARY_KEY_CONSTRAINT_NAME)
-            .doUpdate()
-            .set(CURRENT_VALUE, CURRENT_VALUE.plus(amount))))
-            .onErrorResume(ex -> {
-                LOGGER.warn("Failure when increasing {} {} quota for {}. Quota current value is thus not updated and needs re-computation",
-                    quotaKey.getQuotaComponent().getValue(), quotaKey.getQuotaType().getValue(), quotaKey.getIdentifier(), ex);
-                return Mono.empty();
-            });
+        return updateCurrentValue(quotaKey, amount, IS_INCREASE)
+            .switchIfEmpty(Mono.defer(() -> insert(quotaKey, amount, IS_INCREASE)))
```

Comment:
Two remarks/questions here:
- we loose logs in case of failures
- if the value exists but the update fails for some reason, does it mean it gets wrongly reinit here?

Reply:
> we loose logs in case of failures

I think logs here are unnecessary, the old query already checks on `constraint`, so it will log an error only when has a problem with database. Then not only this method, all Postgres method has the same issue, system issue

> if the value exists but the update fails for some reason, does it mean it gets wrongly reinit here?

It will not reinit, we just handle `switchIfEmpty` not handle `onError`, 
If update fails for some reason, the reactor with throw it